app/main.py

```diff
@@ -1,13 +1,21 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "passed" if not errors else "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(path, linter_report[path])
+            for path in linter_report]
```

Comment:
It can be better formated like

```
return [
    format_single_linter_file(path, linter_report[path])
    for path in linter_report
]
```

Also you can use `items()`to iterate through key and value of your dict instead of getting value from your dict by a key